### PR TITLE
added system design documents

### PR DESCRIPTION
**Pull request by Nikhil-dot-Kumar**
@msrivathsa  @pramurthy @upendrasahu  please review

### CODE REVIEW

File: documentation/design/software_design.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# **Gitops Plugin Design**
+Gitops software will provide ability to monitor git repositories for commits , changes etc and send the data to observability platforms like snappyflow.
+Gitops software will be provided as a standalone binary for different operating system. It will be supporting all major operating systems including linux(major distributions) , windows etc
+
+## **High level diagram**
+![HLD](./software_design_high_level.png)
```
Comment (aswinbh) [inline documentation/design/software_design.md:6]: Create a "images" folder under "design" and move images there

Comment (Nikhil-dot-Kumar) [reply documentation/design/software_design.md]: done

File: documentation/design/input.md [hunk 2]
```diff
@@ -4,6 +4,8 @@ Github audit plugin will be using config.yaml file for taking input from user
```
Comment (aswinbh) [inline documentation/design/input.md:4]: instead of "documentation", let's call the folder "docs"

Comment (Nikhil-dot-Kumar) [reply documentation/design/input.md:4]: done

File: documentation/design/software_design.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# **Gitops Plugin Design**
```
Comment (msrivathsa) [inline documentation/design/software_design.md]: General comments:
     Use present tense in all statement.
     Mention the data structures used
     This design gives Introduction to components.  The actions performed by the component has to be mentioned in separate bullet list.  Though it looks like a low level design, this being a small component, those details are acceptable. If one goes through these statements, it should be clear about the sequence of statements/actions performed by the component.
     This design does not cover the execution flow. For example, if there are 10 jobs, how these are scheduled, and how they get executed.
     Error cases have to be thought through and has to be captured in design.

File: documentation/design/software_design.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+Gitops software will provide ability to monitor git repositories for commits , changes etc and send the data to observability platforms like snappyflow.
```
Comment (msrivathsa) [inline documentation/design/software_design.md]: Use consistent naming. Repo says git audit, here we say Gitops software.

File: documentation/design/software_design.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+Publisher will send the data to target as defined in the input configuration. Target can be anything like elasticsearch , kafka etc
+
+
+## **Important Points**
+1. Software will take user's credentials from environment variable.It will be user's responsibility to add credentials to system environment variables. User credentials can also taken as base64 encoded in config.yaml (less preferred method)
+2. Software will not be fetching old data and will only fetch newer data (from the time software is started)and send it to targets.
+3. Software will be using log files for debugging and informational purpose which will be present at same location as the software binary.
+4. All the builds will be provided as release through github actions.
+
+## **Coding guidelines**
```
Comment (msrivathsa) [inline documentation/design/software_design.md]: Coding guide lines can be a separate document. Need not be part of design.

File: documentation/design/software_design.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+### ***Metric formator***
+Metric formator will be reponsbile for telling the program how to decode each API's response into output. It will consist of json file containing mapping for APIs reponse to ouput format.
+
+### ***Publisher***
+Publisher will send the data to target as defined in the input configuration. Target can be anything like elasticsearch , kafka etc
+
+
+## **Important Points**
+1. Software will take user's credentials from environment variable.It will be user's responsibility to add credentials to system environment variables. User credentials can also taken as base64 encoded in config.yaml (less preferred method)
+2. Software will not be fetching old data and will only fetch newer data (from the time software is started)and send it to targets.
```
Comment (msrivathsa) [inline documentation/design/software_design.md]: Design never says where this state information is maintained.

**Review by msrivathsa — CHANGES_REQUESTED** [review]

File: documentation/design/software_design.md (hunk 3 above)
Comment (msrivathsa) [reply documentation/design/software_design.md]: These are the additional comments.
How to stop the system? When a jobs are in progress, these have to be gracefully completed.

File: documentation/design/software_design.md (hunk 4 above)
Comment (Nikhil-dot-Kumar) [reply documentation/design/software_design.md]: done

File: docs/design/contributing.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+## Coding Style
```
Comment (upendrasahu) [inline docs/design/contributing.md:1]: Keep the file name in capital e.g. CONTRIBUTING.md

File: docs/design/input.md [hunk 8]
```diff
@@ -4,6 +4,8 @@ Github audit plugin will be using config.yaml file for taking input from user
```
Comment (upendrasahu) [inline docs/design/input.md:4]: Comments should be above the field 
## API token in base64 encode format. <REQUIRED/OPTIONAL>, default value: 
access_token: <>

**Review by pramurthy — APPROVED** [review]

**Review by msrivathsa — APPROVED** [review]

File: docs/design/input.md (hunk 8 above)
Comment (Nikhil-dot-Kumar) [reply docs/design/input.md:4]: done

File: docs/design/contributing.md (hunk 7 above)
Comment (Nikhil-dot-Kumar) [reply docs/design/contributing.md]: done

**Pull request merged**